Core/Net/NetFileCache.cs

```diff
@@ -215,6 +272,88 @@ public string GetCachedZip(Uri url, bool test_data = false)
         }
 
         /// <summary>
+        /// Moves the old cache folder to a new one.
+        /// </summary>
+        /// <param name="newPath">The path to move to.</param>
+        /// <param name="move"></param>
+        public bool MoveDefaultCache(string newPath, bool move = false)
+        {
+            if (string.IsNullOrWhiteSpace(newPath))
+            {
+                return false;
+            }
+
+            // Create the new path if it doesn't exists
+            if (!Directory.Exists(newPath))
+            {
+                Directory.CreateDirectory(newPath);
+            }
+
+            // Check that we have list permission
+            try
+            {
+                Directory.GetFiles(newPath);
```

Comment:
This will load the entire list of files in the target directory into memory, which could be a performance hit if there are already a bunch of files there (e.g., if merging multiple separate instance caches). Is there another check we could do instead? What if we just created the test file?